From: =?utf-8?q?Timm_B=C3=A4der?= <mail@baedert.org>
Date: Sat, 17 Mar 2018 13:14:25 +0100
Subject: widget: Only print allocation warnings with --enable-debug

i.e. if G_ENABLE_CONSISTENCY_CHECKS is defined. This makes sense,
considering that this warning is about consistency of internal widget
state.

Origin: upstream, 3.22.30, commit:eb01ba8573bec47e02c8fc4eaf10e2f7cff522dd
---
 gtk/gtkwidget.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gtk/gtkwidget.c b/gtk/gtkwidget.c
index 13421ea..c910e43 100644
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -5981,12 +5981,14 @@ gtk_widget_size_allocate_with_baseline (GtkWidget     *widget,
       gtk_widget_queue_draw (widget);
     }
 
+#ifdef G_ENABLE_CONSISTENCY_CHECKS
   if (gtk_widget_get_resize_needed (widget))
     {
       g_warning ("Allocating size to %s %p without calling gtk_widget_get_preferred_width/height(). "
                  "How does the code know the size to allocate?",
                  gtk_widget_get_name (widget), widget);
     }
+#endif
 
   if (GTK_DEBUG_CHECK (GEOMETRY))
     {
